app/views/unattended/provisioning_templates/user_data/preseed_default_user_data.erb

```diff
@@ -13,9 +13,9 @@ oses:
 
 <%
   # safemode renderer does not support unary negation
-  pm_set = @host.puppetmaster.empty? ? false : true
+  pm_set = (!plugin_present?('foreman_puppet') || @host.puppet.nil?) ? false : true
```

Comment:
This can be similar to the chef check below (and maybe unified into `puppet_enabled`):
```suggestion
  pm_set = @host.respond_to?(:puppetmaster) && @host.puppetmaster.present?
```